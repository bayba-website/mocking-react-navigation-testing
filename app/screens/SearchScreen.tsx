import * as React from 'react';
import {View, Text, Button} from 'react-native';
import {useNavigation} from '@react-navigation/native';

const SearchScreen: React.FC = () => {
  const navigation = useNavigation();
  const goHome = () => navigation.navigate('Home');
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Search Screen</Text>
      <Button title="Home" onPress={goHome} />
    </View>
  );
};

export default SearchScreen;
