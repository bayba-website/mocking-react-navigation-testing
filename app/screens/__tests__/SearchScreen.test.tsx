import React from 'react';
import {Text} from 'react-native';
import {create, act, ReactTestRenderer} from 'react-test-renderer';
import MockNavigation from '../../../mock/mock-navigation';
import SearchScreen from '../SearchScreen';

describe('Search Screen', () => {
  let render: ReactTestRenderer;
  beforeAll(() => {
    act(() => {
      render = create(<MockNavigation component={SearchScreen} />);
    });
  });

  it('should component correctly and matches snapshot', () => {
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should component correctly Search Screen', () => {
    const screen = render.root.findAllByType(Text)[0].props.children;
    expect(screen).toEqual('Search Screen');
  });
});
