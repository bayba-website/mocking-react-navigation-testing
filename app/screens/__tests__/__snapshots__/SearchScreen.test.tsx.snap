// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Screen should component correctly and matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Search Screen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 49,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="MockScreen, tab, 1 of 1"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "selected": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(0, 122, 255)",
                },
                Object {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            MockScreen
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
